dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoInterface.java

```diff
@@ -230,9 +250,10 @@ public Workflow createStubBioworkflow(String repository) {
      * @param repositoryId Repository ID (ex. dockstore/dockstore-ui2)
      * @param existingWorkflow Optional existing workflow
      * @param versionName Optional version name to refresh
+     * @param hardRefresh
      * @return workflow
      */
-    public Workflow createWorkflowFromGitRepository(String repositoryId, Optional<Workflow> existingWorkflow, Optional<String> versionName) {
+    public Workflow createWorkflowFromGitRepository(String repositoryId, Optional<Workflow> existingWorkflow, Optional<String> versionName, boolean hardRefresh) {
```

Comment:
Codacy found an issue: [The method 'createWorkflowFromGitRepository(String, Optional, Optional, boolean)' has an NPath complexity of 288](https://app.codacy.com/gh/dockstore/dockstore/pullRequest?prid=6348842)